legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/ExecutionNodeExecutor.java

```diff
@@ -230,6 +234,20 @@ public Result visit(AllocationExecutionNode allocationExecutionNode)
         {
             result = new ConstantResult(((Map<?, ?>) ((ConstantResult) result).getValue()).get("value"));
         }
+        // TODO HSO: Right place for this? Other streaming result types?
+        if (result instanceof JsonStreamingResult && allocationExecutionNode.realizeInMemory)
```

Comment:
We should consider a few things here:

- What will mean to encapsulate this logic on the `realizeInMemory` method of the Result class?
- If we cannot do that, should we consider the abstraction StreamingResult rather than this specific implementation?
- Last, should the ConstatResult be a string or a byte array?

We dont not need answers to all this, and we might need to take an approach and stick to it (like implement the realize to memory and do exactly this inside there)  I understand that forces a particular JSON serialize format, but as mention, we will need a version, and you already will be creating precedent even if we go with the current proposal on this MR.